### PR TITLE
Update to cmake 3.9 - latest version (should be used with libuv and e…

### PR DESCRIPTION
**Pull request by JPeterMugaas**
…xpat from previous PR requests)

mingw-w64-ncurses -  6.0  20170715 - Update to latest version
mingw-w64-libarchive - 3.3.2 - Update to latest version and update patch for bcrypt.  The logic is a little better than what was in the source-code
mingw-w64-cmake - 3.9.0 - Update to latest version

### CODE REVIEW

**Comment by holgern** [comment]
mingw-w64-crt-git is missing as dependency, therefore building mingw-w64-libarchive fails

**Comment by mati865** [comment]
@holgern it is in `*-toolchain` groups

**Comment by holgern** [comment]
i agree it is installed on appveyor, but it seems that is not installed on teaCi.

I tried to compile libarchive without mingw-w64-crt-git and i was getting the same error as seen on teaCi.

**Comment by mati865** [comment]
@holgern it's installed but whole Tea is outdated for over a year.

**Comment by Alexpux** [comment]
@JPeterMugaas your patches have format errors. Fix them

File: mingw-w64-cmake/0004-Implement-Qt5-static-plugin-support.patch [hunk 1]
```diff
@@ -35,23 +35,22 @@ index f11825083e..e8a07361c0 100644
...
 +++ b/Source/cmQtAutoGeneratorInitializer.cxx
-@@ -26,6 +26,7 @@
- #include <cmConfigure.h>
- #include <cmsys/FStream.hxx>
- #include <cmsys/RegularExpression.hxx>
+@@ -25,6 +25,7 @@
+ 
+ #include "cmConfigure.h"
+ #include "cmsys/FStream.hxx"
 +#include <cmGeneratedFileStream.h>
```
Comment (Alexpux) [inline mingw-w64-cmake/0004-Implement-Qt5-static-plugin-support.patch]: Seems wrong indentation

File: mingw-w64-cmake/0004-Implement-Qt5-static-plugin-support.patch [hunk 2]
```diff
@@ -110,47 +109,22 @@ index 825eba0e1d..61652b80b1 100644
 +      }
 +    }
 +  }
```
Comment (Alexpux) [inline mingw-w64-cmake/0004-Implement-Qt5-static-plugin-support.patch]: Previous 7 lines are wrong.

File: mingw-w64-cmake/0004-Implement-Qt5-static-plugin-support.patch [hunk 3]
```diff
@@ -110,47 +109,22 @@ index 825eba0e1d..61652b80b1 100644
...
+diff --git a/Source/cmTarget.cxx.orig b/Source/cmTarget.cxx
+index c95a3ca..9ec1995 100644
+--- a/Source/cmTarget.cxx.orig
 +++ b/Source/cmTarget.cxx
-@@ -245,6 +245,7 @@ cmTarget::cmTarget(std::string const& name, cmStateEnums::TargetType type,
+@@ -247,6 +247,7 @@ cmTarget::cmTarget(std::string const& name, cmStateEnums::TargetType type,
      this->SetPropertyDefault("AUTOMOC", CM_NULLPTR);
      this->SetPropertyDefault("AUTOUIC", CM_NULLPTR);
      this->SetPropertyDefault("AUTORCC", CM_NULLPTR);
 +    this->SetPropertyDefault("AUTOSTATICPLUGINS", CM_NULLPTR);
```
Comment (Alexpux) [inline mingw-w64-cmake/0004-Implement-Qt5-static-plugin-support.patch]: Wrong indentation

**Pull request merged**